06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js

```diff
@@ -0,0 +1,276 @@
+   * @memberOf InvertedIndexUtility
+   */
+  static filterIndexed(query, indexedFile) {
+    const result = [];
+    query.forEach((term) => {
+      indexedFile.forEach((index) => {
+        if (term === index[0]) {
+          result.push(index);
+        }
+      });
+    });
+    return result;
+  }
+
+  /**
+   * sanitizeQuery method removes unwanted terms in the query string
+   * and also converts the query string into an array of each unique term
+   * @static
+   * @param {String} query - The terms to search for the index
+   * @returns {Array} - The already formatted query
+   * @memberOf InvertedIndexUtility
+   */
+  static sanitizeQuery(query) {
+    query = query.toLowerCase()
+    .replace(/-/g, ' ')
+    .replace(/[^A-z\s]/g, '')
+    .split(' ');
+    let result = [];
+    query.forEach((word) => {
+      if (word.trim() !== '') {
+        result.push(word);
+      }
+    });
+    result = this.removeDuplicateWords(result);
+    return result;
+  }
+
+  /**
+   * This method remove all the duplicate words in an Array
+   * @param {Array} words - an array of words
+   * @returns {Array} - a new array with no duplicate words
+   */
+  static removeDuplicateWords(words) {
+    const uniqueWords = [];
+    for (let i = 0; i < words.length; i += 1) {
+      if (words.lastIndexOf(words[i]) === i) {
+        uniqueWords.push(words[i]);
+      }
+    }
+    return uniqueWords;
+  }
+
+   /**
+   * This methods gets all the tokens in a text
+   * @param {String} text - A text of words
+   * @returns {Array} - An array consisting of all the tokens in the text
+   */
+  static tokenize(text) {
+    return text.trim().replace(/-/g, ' ')
+    .replace(/[^A-z\s]/g, '')
+    .toLowerCase()
+    .split(' ');
+  }
+
+  /**
+   * @static
+   * @param {Array} words - An array of all the words in the text of the book
+   * @param {Array} wordsInEachBook - An Array of all the words in each book
+   * @param {Array} bookTitles - An array that consist of a sub array
+   * of all the words in each book
+   * @returns {Array} - An array consisiting of a sub array with each word
+   * and booleans to indicate their occurence in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static constructIndex(words, wordsInEachBook, bookTitles) {
+    const checked = this
+    .checkOccurenceOfWords(words, wordsInEachBook);
+    const booksLength = bookTitles.length;
+    const booksChecked = this
+    .getWordOccurenceForEachBook(checked, booksLength);
+    const indexConstructed = this
+    .mapWordsWithOccurence(words, booksChecked);
+    return indexConstructed;
+  }
+
+  /**
+   * This method gets the title of each book in a file
+   * @static
+   * @param {Array} file - The content of the given file
+   * @returns {Array} - The Title of each book in the file
+   * @memberOf InvertedIndexUtility
+   */
+  static getTitlesOfEachBook(file) {
+    const titleList = [];
+    for (let i = 0; i < file.length; i += 1) {
```

Comment:
This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)